crates/oxc_linter/src/rules/typescript/no_unsafe_declaration_merging.rs

```diff
@@ -0,0 +1,190 @@
+use oxc_ast::{ast::BindingIdentifier, AstKind};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::SymbolId;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(no-unsafe-declaration-merging): Unsafe declaration merging between classes and interfaces.")]
+#[diagnostic(severity(warning))]
```

Comment:
Shall we add a `#[help]` here? It would be helpful for the user to know why it is unsafe. The doc says

"The TypeScript compiler doesn't check whether properties are initialized, which can cause lead to TypeScript not detecting code that will cause runtime errors."